### ISSUE TITLE
[Enhancement] filter across multiple pages

### ISSUE THREAD

**Issue by barredterra** [knadh/dirmaker#5]
Imagine we have a list of 100 entries in a category, with a page length of 10. Currently, we can only apply filters to the 10 visible entries. What I would expect is that filters are applied on all entries and the matches are paginated.

To put it differently, I can only use one feature effectively, either filters or pagination. Using both at the same time currently leads to unexpected behavior.

**Comment by knadh** [comment]
Yep, the filters are in-page currently. To be able to search the entire directory, it's necessary to bundle some sort of a JS indexing+search lib. We can consider bundling one while making it on/off in the config.
